x-pack/plugins/actions/server/types.ts

```diff
@@ -14,8 +15,31 @@ export interface SavedObjectReference {
   id: string;
 }
 
+export interface Services {
+  callCluster: any;
+  savedObjectsClient: SavedObjectsClientContract;
+  log: (tags: string | string[], data?: string | object | (() => any), timestamp?: number) => void;
+}
+
 export interface ActionsPlugin {
   registerType: ActionTypeRegistry['register'];
   listTypes: ActionTypeRegistry['list'];
-  fire: ({ id, params }: { id: string; params: Record<string, any> }) => Promise<any>;
+  fire: ({ id, params }: { id: string; params: Record<string, any> }) => Promise<void>;
+}
+
+export interface ActionTypeExecutorOptions {
+  services: Services;
+  actionTypeConfig: Record<string, any>;
+  params: Record<string, any>;
+}
+
+export interface ActionType {
+  id: string;
+  name: string;
+  unencryptedAttributes?: string[];
+  validate?: {
```

Comment:
sort of feels like this should be named `validator`.  Also, seems weird to call params `params` but cofig `actionTypeConfig`.  Can we shorted this to just `config` everywhere?  So, Actions have a config, which was passed to ActionType when creating the Action, and params are what is sent to the Action when it's going to be executed.